contracts/BSCValidatorSet.sol

```diff
@@ -1294,9 +1298,10 @@ contract BSCValidatorSet is IBSCValidatorSet, System, IParamSubscriber, IApplica
             _misdemeanor(validator);
         }
 
-        // step 3: modify global storage
-        --numOfMaintaining;
-        validatorExtraSet[index].isMaintaining = false;
+        // step 4: clear isMaintaining info
```

Comment:
if this is the case, we can move `validatorExtraSet[index].isMaintaining = false;` up without condition.